api/src/play_asset_pack_manager.gd

```diff
@@ -99,3 +99,35 @@ func get_pack_location(pack_name : String) -> PlayAssetPackLocation:
 	if query_dict == null:
 		return null
 	return PlayAssetPackLocation.new(query_dict)
+
+# -----------------------------------------------------------------------------
+# Returns the location of all installed asset packs. More specifically, a 
+# Dictionary, where for each entry, the key is the asset pack anme and value is 
+# the corresponding PlayAssetLocation object.
+# -----------------------------------------------------------------------------
+func get_pack_locations() -> Dictionary:
+	var return_dict = Dictionary()
+	var raw_dict = _plugin_singleton.getPackLocations()
+	
+	# convert inner dictionaries in raw_dict to PlayAssetLocation objects
+	for key in raw_dict.keys():
+		return_dict[key] = PlayAssetPackLocation.new(raw_dict[key])
+	
+	return return_dict
+
+# -----------------------------------------------------------------------------
+# Cancel a asset pack request specified with pack name, true if success. Note: 
+# Only active downloads can be canceled.
+# -----------------------------------------------------------------------------
+func cancel_asset_pack_request(pack_name : String) -> bool:
+	var raw_dict = _plugin_singleton.cancel([pack_name])
+	var updated_asset_pack_states : PlayAssetPackStates = PlayAssetPackStates.new(raw_dict)
+	
+	# return false if matching pack_name is found in updated PlayAssetPackStates
```

Comment:
Can I double check on the behavior of canceling some non-existing packs?  Currently I'm treating it as if the returned AssetPackStates object won't contain an AssetPackState with matching packName.

Reply:
I believe you're handling this correctly. AFAIK the returned AssetPackStates object returned AssetPackStates object won't contain an AssetPackState with matching packName.